### ISSUE TITLE
Feature request: Use Result alias in extract-to-function assist

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-analyzer#12307]
Currently, when I use `extract to function` on some fallible code in a function that returns `syn::Result<_>`, `anyhow::Result<_>` or a similar type alias for `Result<_, SomeError>`, the generated function uses the expanded form of that type alias. It would be nice if it used the same result alias as the function it is extracted from.

**Comment by flodiebold** [comment]
That's pretty much #1666.

**Comment by jplatte** [comment]
Right. If you don't expect a fix for this specifically to be easier than a full fix for #1666, feel free to close.
